packages/legend-graph/src/graphManager/protocol/pure/v1/transformation/pureProtocol/serializationHelpers/V1_TestSerializationHelper.ts

```diff
@@ -145,10 +160,23 @@ const V1_deserializeAssertionStatus = (
       return deserialize(V1_assertPassModelSchema, json);
     case V1_AssertionStatusType.EQUAL_TO_JSON_ASSERT_FAIL:
       return deserialize(V1_equalToJsonAssertFailModelSchema, json);
-    default:
+    default: {
+      const extraAssertionStatusProtocolDeserializers = plugins.flatMap(
+        (plugin) =>
+          (
+            plugin as Testable_PureProtocolProcessorPlugin_Extension
+          ).V1_getExtraAssertionStatusProtocolDeserializers?.() ?? [],
+      );
+      for (const deserializer of extraAssertionStatusProtocolDeserializers) {
+        const assertionStatusProtocol = deserializer(json, plugins);
+        if (assertionStatusProtocol) {
+          return assertionStatusProtocol;
+        }
+      }
       throw new UnsupportedOperationError(
         `Can't deserialize assertion status of type '${json._type}'`,
```

Comment:
Would you mind updating the message to similar format in places wherever you added the extension mechanism in the code.
`Can't deserialize assertion status of type '${json._type}': no compatible serializer available from the plugins`

Reply:
Sure, changed!